src/Core/Manifest/VersionProvider.php

```diff
@@ -43,6 +44,11 @@ class VersionProvider
      */
     public function getVersion()
     {
+        $key = sprintf('%s-%s', $this->getComposerLockPath(), 'all');
```

Comment:
API on VersionedProvider allows changing the composer.lock file, so we need to cater for this in the cache key - https://github.com/silverstripe/silverstripe-framework/blob/4/src/Core/Manifest/VersionProvider.php#L170

In practice this would only be used for unit tests which change the path https://github.com/silverstripe/silverstripe-framework/blob/4/tests/php/Core/Manifest/VersionProviderTest.php#L22

If we don't include the composer lock path in the cache key here, tests will not pass

Reply:
Also I don't think the lock file hash is required, as it's a fair to assume that whenever a composer lock file is updated, flush=1 will be run afterwards.  Using the file path instead of the the hash contents saves a file read operation

Reply:
Fair. I recommended that without the context that `VersionProvider` allows changing the composer.lock file, so I made an incorrect assumption about the purpose of the caching.